### ISSUE TITLE
Can SpineOMatic 'remember' chosen parameters for gridline & marginless templates&layouts?

### ISSUE THREAD

**Issue by DiederikL** [ExLibrisGroup/spineomatic-cloudapp#98]
Hello,

As it is my first time posting an issue concerning SpineOMatic cloud app for Alma, I want to shortly introduce myself: my name is Diederik Lanoye and I am a business consultant at LIBIS, the ICT department of the University Library of Leuven (Belgium). I am the successor of Vangelis Palaskas who posted a couple of issues in 2021-2022.

We are now implementing SpineOMatic throughout our library network. The new tool (coming from Alma Label tool) is highly appreciated. However, one question keeps returning. (Almost) all of our templates are designed to be printed with the option 'Enable marginless templates and layouts' on. This option makes the design of templates and layouts much easier. So our colleagues who wish to print labels have to ensure that this option is activated. The problem is that unlike their choice of template and layout, SpineOMatic does not appear to have a 'memory' for this parameter so each time they open SpineOMatic to print a new label, we have to 're-activate' the option. 

Would it be possible to develop such an 'option memory'?

Best regards, 

Diederik Lanoye

**Comment by mgobat** [comment]
@sthb13, since your institution contributed the enhancement for the "Enable marginless templates and layouts" option, what do you think of this suggestion? Thank you.

**Comment by sthb13** [comment]
Yes, that would be very nice to have as most, if not all our libraries have switched to the new templates and would save the users one click that is easy to forget. I don't have the time right now to implement this, not until March at the earliest. So if you @mgobat could do that, it would be very much appreciated from our side.

Thanks and best regards, 
Skúli Þór
CIL Consortium of Icelandic Libraries

**Comment by mgobat** [comment]
@sthb13, thanks for your input.

@sthb13, @DiederikL, from your perspective, does it make more sense to include the option in the template or in the layout?

**Comment by DiederikL** [comment]
Hi @mgobat and @sthb13 ,

Many thanks for your quick reaction. Actually, the solution I was thinking about was a little bit different. 

In the current version of SpineOMatic you can choose to print marginless or not with a slider button on the SpineOMatic screen where you also select the layout and template to be used. If you want to print marginless you activate the option by clicking on the slider button (the same for 'Print gridlines'). If you close down SpineOMatic and at a later moment return to print new labels, the layout and template you selected on your last session are still there but even if you left SpineOMatic with marginless printing activated, this option (like the 'Print gridlines') is deactivated. 

![image](https://user-images.githubusercontent.com/100340483/213471696-fcc01870-4c04-495f-99d2-2cdca600c24e.png)

It would be great if the status of both options (activated or not) would only change if you change them manually so that if marginless templates were activated when you left your previous SpineOMatic session, the slider would still be in activated position when returning.

But obviously, a development where you could determine whether a label is to be printed marginless or not when designing layout and template would also do. In that case, I think it would make more sense to include this in the layout design rather than template design as margin and other dimension settings are part of layout design.

Best regards, 

Diederik Lanoye

**Comment by mgobat** [comment]
@DiederikL, I believe saved values can only exist in the configuration and settings interfaces in Cloud Apps. I'll check.

**Comment by sthb13** [comment]
I was thinking this like @DiederikL as well but i'm not sure about the implementation details of that. Having a setting in the templates or layouts that would then be reflected in the print interface for the end user, i.e. the switch would be on if the template or layout was intended for marginless layout, would be perfectly fine for our use case.

**Comment by mgobat** [comment]
@DiederikL, @sthb13, it looks like this can be done using the Cloud App Store Service. The Cloud App Store Service is already used for saving the last selected template and layout.

**Comment by DiederikL** [comment]
@mgobat That sounds great. Do you have any idea about the time you will need to implement this? Medio March I will organize two online workshops to start the rollout of SpineOMatic throughout the KU Leuven Consortium. It would be great if this enhancement would be available at that time.

**Comment by mgobat** [comment]
@DiederikL, I'm targeting February for the next release. This implementation will save the setting of the two toggle switches just as it currently saves the user's previous selection for the template and layout. See https://developers.exlibrisgroup.com/cloudapps/docs/api/store-service/ for reference.

**Comment by sthb13** [comment]
That's good news, thank you @mgobat

**Comment by mgobat** [comment]
Available in release v1.3.1.

**Comment by DiederikL** [comment]
Works perfect! Many thanks @mgobat
